e2e_tests/tests/cluster/test_users.py

```diff
@@ -367,6 +367,25 @@ def test_login_as_non_existent_user(clean_auth: None) -> None:
     assert child.exitstatus != 0
 
 
+@pytest.mark.e2e_cpu
+def test_login_det_pass_environment(clean_auth: None) -> None:
+    creds = create_test_user(ADMIN_CREDENTIALS, True)
+
+    try:
+        os.environ["DET_PASS"] = creds.password
```

Comment:
nit: if this were to fail, your finally would also fail.  So technically, it should be:
```python
# setup before try
os.environ["DET_PASS"] = creds.password
try:
    # do stuff
    ...
finally:
    # cleanup in finally
    del os.environ["DET_PASS"]
```